### ISSUE TITLE
Add `onDidResize` callback

### ISSUE THREAD

**Issue by martonlanga** [DevbookHQ/splitter#3]
Would be nice to have this in order to store the user's preferences.

Api could look something like this:
```diff
interface SplitProps {
  direction: SplitDirection;
  minWidth?: number; // In pixels.
  minHeight?: number; // In pixels.
  initialSizes?: number[]; // In percentage.
+ onSizesChanged?: (sizes: number[]) => void; // Called when sizes change
  gutterClassName?: string;
  draggerClassName?: string;
  children?: React.ReactNode;
}
```

**Comment by mlejva** [comment]
This is prepared on the `develop` branch. 

There is one bug preventing the release though that causes Splitter's children to reset their width. It's explained [here](https://github.com/DevbookHQ/splitter/blob/develop/example/src/OnDidResize/index.tsx#L11).

Right now, I don't have time to look into this but if you fix the problematic code I'm more than happy to accept the PR.

**Comment by mlejva** [comment]
Implemented in the `v1.0.4`. Check an out example [here](https://github.com/DevbookHQ/splitter/blob/master/example/src/OnDidResize/index.tsx).
